### PR TITLE
Generate informational header at top of YAML file.

### PR DESCRIPTION
**Pull request by mwhittaker**
This PR changes `weaver kube deploy` to add an informational header at the top of every generated YAML file. It looks like this:

```yaml
# This file was generated by "weaver kube" version (devel) for the following
# application:
#
#     app: collatz
#     version: 79f011d4
#     components:
#     - github.com/ServiceWeaver/weaver/Main
#     - github.com/ServiceWeaver/weaver/examples/collatz/Even
#     - github.com/ServiceWeaver/weaver/examples/collatz/Odd
#     listeners:
#     - collatz
#
# This file contains the following resources:
#
#     1. A Deployment for every group of components.
#     2. A HorizontalPodAutoscaler for every Deployment.
#     3. A Service for every listener.
#     4. Some Roles and RoleBindings to configure permissions.
#
# To deploy these resources, run:
#
#     kubectl apply -f /tmp/kube_79f011d4.yaml
#
# To view the deployed resources, run:
#
#     kubectl delete all --selector=version=79f011d4
...
```

### CODE REVIEW

File: internal/impl/kube.go [hunk 1]
```diff
@@ -489,6 +495,82 @@ func GenerateYAMLs(image string, app *protos.AppConfig, depId string, cfg *KubeC
...
+#     - {{.}}
+      {{- end}}
+#     listeners:
+      {{- range .Listeners}}
+#     - {{.}}
+      {{- end}}
+#
+# This file contains the following resources:
+#
+#     1. A Deployment for every group of components.
```
Comment (rgrandl) [inline internal/impl/kube.go:516]: would it be possible to list how these groups of components look like?

File: internal/impl/kube.go [hunk 2]
```diff
@@ -489,6 +495,82 @@ func GenerateYAMLs(image string, app *protos.AppConfig, depId string, cfg *KubeC
...
+	header := template.Must(template.New("header").Parse(`# This file was generated by "weaver kube" version {{.ToolVersion}} for the following
+# application:
+#
+#     app: {{.App}}
+#     version: {{.Version}}
+#     components:
+      {{- range .Components}}
+#     - {{.}}
+      {{- end}}
+#     listeners:
```
Comment (rgrandl) [inline internal/impl/kube.go:509]: do you think it makes sense to mention the name of the component/group for the listener?

File: internal/impl/kube.go [hunk 3]
```diff
@@ -489,6 +495,82 @@ func GenerateYAMLs(image string, app *protos.AppConfig, depId string, cfg *KubeC
 	return nil
 }
 
+// header returns the informational header at the top of a generated YAML file.
+func header(app *protos.AppConfig, cfg *KubeConfig, depId, filename string) (string, error) {
+	header := template.Must(template.New("header").Parse(`# This file was generated by "weaver kube" version {{.ToolVersion}} for the following
```
Comment (rgrandl) [inline internal/impl/kube.go:500]: I was thinking whether long term we should also provide more k8s stuff. E.g., for every deployment, stuff like cpu/memory request/limit, env variables set for the pods, min/max #replicas, volume mounts. Maybe for HPA, which rules apply. If someone takes a quick look at the header can get a good understanding of the high-level things that would happen if they deploy the yaml.

**Review by rgrandl — APPROVED** [review]
Nice. Thanks Michael

File: internal/impl/kube.go (hunk 2 above)
Comment (mwhittaker) [reply internal/impl/kube.go:511]: Good idea, done.

File: internal/impl/kube.go (hunk 1 above)
Comment (mwhittaker) [reply internal/impl/kube.go:518]: Good idea, done!

File: internal/impl/kube.go (hunk 3 above)
Comment (mwhittaker) [reply internal/impl/kube.go:500]: Sounds like a good idea to me! I'll do this in a future PR.

**Review by mwhittaker — COMMENTED** [review]
Thanks Robert!

**Pull request merged**